pkg/interfaces/protocolprofile.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2024 IOTech Ltd
+
+package interfaces
+
+import (
+	sdkModels "github.com/edgexfoundry/device-sdk-go/v3/pkg/models"
+	model "github.com/edgexfoundry/go-mod-core-contracts/v3/models"
+)
+
+// ProfileScan is a low-level device-specific interface implemented
+// by device services that support dynamic profile scan.
+type ProfileScan interface {
+	// ProfileScanFunc triggers specific device to discover device profile.
+	ProfileScanFunc(req sdkModels.ProfileScanRequest) (model.DeviceProfile, error)
```

Comment:
```suggestion
	ProfileScan(req sdkModels.ProfileScanRequest) (model.DeviceProfile, error)
```
It's fine to use the same name as the interface, because will merge this func to the ProtocolDriver in the next release